lib/asset/AssetGenerator.js

```diff
@@ -144,6 +146,19 @@ class AssetGenerator extends Generator {
 							contentHash
 						}
 					);
+					if (assetModulePublicPath) {
+						if (typeof assetModulePublicPath === "function") {
+							assetModulePublicPath = assetModulePublicPath(
+								filename,
+								module.matchResource || module.resource,
+								runtimeTemplate.compilation.compiler.context
+							);
+						}
+						assetModulePublicPath = JSON.stringify(assetModulePublicPath);
```

Comment:
```suggestion
						publicPath = JSON.stringify(assetModulePublicPath);
```

Use a few variable for that.